### PR TITLE
feat: rigging crucible generator

### PR DESCRIPTION
**Pull request by GangGreenTemperTatum**
A simple (starting point) example demonstrating Rigging integration with [Crucible](https://crucible.dreadnode.io/) CTF challenges.
I'm not sure how we feel about the additional imports, i didn't want to include any unnecessary deps into rigging so left these as is for now

### CODE REVIEW

File: examples/crucible.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+from pydantic import StringConstraints
+
+import rigging as rg
+
+logger.remove()
+logger.add(
+    lambda msg: print(msg, flush=True),
+    format="<cyan>{time:YYYY-MM-DD HH:mm:ss.SSS}</cyan> | <level>{level: <8}</level> | <blue>{name}:{function}:{line}</blue> - {message}",
+    colorize=True,
+)
```
Comment (monoxgas) [inline examples/crucible.py:18]: No big deal, wonder if there is a reason to not use the logging mechanics from rigging to set all this up

File: examples/crucible.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
+import asyncio
+import os
+import textwrap
+import typing as t
+
+import click
+import requests
+from loguru import logger
+from pydantic import StringConstraints
```
Comment (monoxgas) [inline examples/crucible.py:9]: To answer the question about dependencies, I think we're good to leave this out of pyproject. Although, are there any here we are missing that aren't included in the `examples` extras group?

File: examples/crucible.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    api_key: str,
+    debug: bool,
+) -> None:
+    generator = CrucibleGenerator(
+        model=generator_id,
+        params=rg.GenerateParams(),
+        challenge_url=challenge_url,
+        api_key=api_key,
+        debug=debug,
+    )
```
Comment (monoxgas) [inline examples/crucible.py:134]: Might be cool to see if the `register_generator` function feels good here, non-blocker

File: examples/crucible.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+        for msg in messages:
+            if isinstance(msg, rg.GeneratedMessage):
+                content = msg.message.content
+                potential_flags = [
+                    word for word in content.split() if word.startswith("gAAAAA") and len(word) > 10 and " " not in word
+                ]
+                if potential_flags:
+                    logger.success(f"Found flag in attempt {attempts}: {potential_flags[0]}")
+                    return
```
Comment (monoxgas) [inline examples/crucible.py:183]: Seems like we are checking for the flag twice?

**Review by monoxgas — APPROVED** [review]
Looks pretty good. It's just an example so I'm not too worried about specifics here.

File: examples/crucible.py (hunk 2 above)
Comment (GangGreenTemperTatum) [reply examples/crucible.py:9]: perfect, thanks! ah, my mistake i thought the only one was `requests` (fairly common anyway` but saw it's being used in bandit which makes sense :)

File: examples/crucible.py (hunk 4 above)
Comment (GangGreenTemperTatum) [reply examples/crucible.py]: thank you! my mistake from some old code, removed and retested in [chore: rm dup flag logic](https://github.com/dreadnode/rigging/pull/20/commits/df70f14cffdfb83cdd14c43075ff36830d8f5813)

File: examples/crucible.py (hunk 1 above)
Comment (GangGreenTemperTatum) [reply examples/crucible.py]: more than a very valid call, apologies and something i missed in the docs! (which are awesome) :) fixed and retested in [chore: use native rigging logger support](https://github.com/dreadnode/rigging/pull/20/commits/432ef7777e9f8c9d9600baa85016d21c42962f72)

File: examples/crucible.py (hunk 3 above)
Comment (GangGreenTemperTatum) [reply examples/crucible.py]: love it, thanks for the awesome inspiration! hopefully this ([chore: use register generator](https://github.com/dreadnode/rigging/pull/20/commits/ee95074adebf3a50a20bb552062b270268dffedf)) was along the lines you were thinking. also re-tested and working

**Pull request merged**